lumicks/pylake/population/dwelltime.py

```diff
@@ -544,8 +583,100 @@ def exponential_mixture_log_likelihood(params, t, min_observation_time, max_obse
     return -np.sum(log_likelihood)
 
 
+def _handle_amplitude_constraint(
+    n_components, params, fixed_param_mask
+) -> Tuple[np.ndarray, Union[Dict, Tuple], np.ndarray]:
+    """Determines how many amplitudes actually need fitting.
+
+    For a single-component model the amplitude is 1 by definition. For an N-component model, where
+    N - 1 amplitudes are fixed, the free amplitude is determined by the constraint. For models that
+    have more than one free amplitude, the amplitudes are constrained to sum to 1.
+
+    Parameters
+    ----------
+    n_components : int
+        number of components in the mixture model
+    params : array_like
+        model parameters
+    fixed_param_mask : array_like
+        logical mask of fixed parameters
+
+    Returns
+    -------
+    Tuple[np.ndarray, Union[Dict, Tuple], np.ndarray]
+        This function returns the mask of parameters to be fitted, amplitude constraint function
+        and updated parameter vector (forced to be consistent with the constraint). It was split
+        out from `_exponential_mle_optimize` to reduce cognitive load.
```

Comment:
```suggestion
    Tuple[np.ndarray, Union[Dict, Tuple], np.ndarray]
        Tuple containing: mask of parameters to be fitted, amplitude constraint function
        and updated parameter vector (forced to be consistent with the constraint). 
```

I would move the final sentence into the main description, something like
```
This logic was split out from `_exponential_mle_optimize` to reduce cognitive load.
```
although I wonder if it's a good idea to include "history" in the docstring. Not a big deal since it's internal, but might be better left as additional info in the commit message?

Reply:
Yeah, I will drop it from the docstring. No need to carry that history with.